### DISCUSSION TITLE
Flying multiple crazyflie with Loco Positioning Node and Crazyswarm

### DISCUSSION THREAD

**Discussion by uhvardhan** [USC-ACTLab/crazyswarm#562] [Q&A] [answered]
Hi, 

Since I do not have a motion capture system in the lab yet, I am currently using Loco Positioning System. I have made any and all changes mentioned in the Installation and Configuration pages of crazyswarm. 1 crazyflie flies perfectly. I am now trying to fly 2:

This is how I set-up the crazyflies.yaml file. A single crazyradio PA is being used only
![Screenshot from 2021-12-15 22-36-39](https://user-images.githubusercontent.com/25361013/146303813-298b21f2-740b-4c72-b6a3-8aebb778e8e4.png)

Now, when I run the hover_launch file, RVIZ never actually gets the approximate position of the 2 crazyflies - They keep changing every second. I do observe a similar behavior when I run cfclient. Never getting the position of the 2 crazyflies. 

Does crazyswarm work with Loco Positioning Node? Because when I try to fly 2 - it crashes - does some weird stuff in half a second and crashes. 

Thank you

**Comment by knmcguire** [comment]
hi! Do you mean with changing every second in RVIZ, that the position is not converging? You could check the console tab / crazyswarm read out with the EKF as it should give a warning of:
`ESTKALMAN: State out of bounds, resetting`

**Reply by uhvardhan** [reply]
Yes. The position is not converging. 

It does give that warning "State out of bounds, resetting"

**Comment by whoenig** [comment]
Crazyswarm supports LPS by setting the motion_capture_type to none. You still need to configure your drones correctly. Which LPS mode are you using and do you have pose_logging enabled?

**Reply by uhvardhan** [reply]
I am pretty sure I have configured the launch file correctly. I have motion capture set to none and I have enabled pose logging and I have been using TWR.

**Reply by uhvardhan** [reply]
Although there is one variable I am not sure about:
motion_capture_host_name is currently set to "vicon". Should I make it None?

**Reply by whoenig** [reply]
> motion_capture_host_name is currently set to "vicon". Should I make it None?

No, that variable is not being used if the motion capture type is none.

**Reply by uhvardhan** [reply]
I just read that TWR doesn't work with multiple crazyflies.

I should change the LPS mode to TDOA2 or TDOA3? Any suggestions as to which one to choose?

**Reply by whoenig** [reply]
TDOA3 is the most recommended/tested version.

**Reply by uhvardhan** [reply]
Thank you! Going to try it out now.

**Comment by uhvardhan** [comment, marked as answer]
LPS mode was initially "TWR". Once I changed the mode to "TDOA3" - I am able to resolve the issue and fly multiple crazyflies. 

Thank you.
